sphinx/domains/std.py

```diff
@@ -544,12 +544,10 @@ class StandardDomain(Domain):
         'objects': {},          # (type, name) -> docname, labelid
         'labels': {             # labelname -> docname, labelid, sectionname
             'genindex': ('genindex', '', _('Index')),
-            'modindex': ('py-modindex', '', _('Module Index')),
```

Comment:
With this change, reference title will be changed because `sphinx.domains.python:ModuleIndex.name` is "Python Module Index". But I don't have opinion for that.